app.js

```diff
@@ -1 +1,73 @@
 'use strict';
+var surveyLength = 3;
+var randomProduct = [0,1,2,];
+var currentProducts = [];
+//create var to hold product objects
+var productList = [
+  new AddProduct('Boots', 'img/boots.jpg'),
+  new AddProduct('Chair', 'img/chair.jpg'),
+  new AddProduct('Scissors', 'img/scissors.jpg'),
+  new AddProduct('Water Can', 'img/water-can.jpg'),
+  new AddProduct('Wine Glass', 'img/wine-glass.jpg')
+];
+
+//create constructor function to build product objects
+//prototype any consistent values i.e. views, votes -> set = 0
+AddProduct.prototype.selectedCount = 0;
+AddProduct.prototype.viewedCount = 0;
+
+function AddProduct(
+  name,
+  image
+) {
+  this.name = name;
+  this.image = image;
+}
+
+function runSurvey() {
+  for (var i = 0; i < 3; i++) {
+    currentProducts[i] = Math.floor(Math.random() * productList.length);
+  }
+  for (var i = 0; i < 2; i++) {
+    var allProductsNew = false;
+    while (!allProductsNew) {
+
+      getThreeNewProducts();
+      checkAllProductsNew(allProductsNew);
```

Comment:
This might get a little easier to read/understand if you could do:

```javascript
while (!checkAllProductsNew()) {
  getThreeNewProducts();
}
```

This loop eventually terminates, but you'll do less work if you specifically look for duplicate values and generate individual replacements, rather than generating a full set each time.